### PR TITLE
GIX-1005: Improve Token Amount Class

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Upgrade nns-js.

# Changes

* Fix usage of `TokenAmount` factory functions to pass always `token` parameter.
* Remove casting when using `fromNumber`.

# Tests

* Fixed broken tests.

### CODE REVIEW

File: frontend/src/lib/api/sns-ledger.api.ts [hunk 1]
```diff
@@ -34,7 +34,8 @@ export const getSnsAccounts = async ({
     principal: identity.getPrincipal(),
     balance: TokenAmount.fromE8s({
       amount: mainBalanceE8s,
-      token: mapOptionalToken(metadata),
+      // TODO: Decide what we do when there is no token in an SNS project
```
Comment (peterpeterparker) [inline frontend/src/lib/api/sns-ledger.api.ts:37]: should we track this with a jira issue? sounds kind of like an essential question to me

File: frontend/src/lib/modals/sns/SwapModal/ParticipateSwapModal.svelte [hunk 2]
```diff
@@ -92,7 +94,7 @@
         account: sourceAccount,
         // If we made it here, we know that the amount is valid
         // TODO: change fromNumber return type
```
Comment (peterpeterparker) [inline frontend/src/lib/modals/sns/SwapModal/ParticipateSwapModal.svelte:96]: we still need this todo or it is actually solve with this change?

File: frontend/src/lib/api/sns-ledger.api.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/lib/api/sns-ledger.api.ts:37]: sure, I added it as a comment https://dfinity.atlassian.net/browse/GIX-1045

File: frontend/src/lib/modals/sns/SwapModal/ParticipateSwapModal.svelte (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/lib/modals/sns/SwapModal/ParticipateSwapModal.svelte:96]: Good point, I can remove this comment

**Comment by lmuntaner** [comment]
`NewTransaction` uses ICPToken as default, but we might want to change that.

Worth checking with Mischa and the rest of the team what should we do then.

File: frontend/src/lib/api/sns-ledger.api.ts (hunk 1 above)
Comment (peterpeterparker) [reply frontend/src/lib/api/sns-ledger.api.ts]: Thanks

**Comment by peterpeterparker** [comment]
It's more a technical question than UX I would say but sure should discuss it.

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

**Pull request merged**